sycl/doc/extensions/experimental/sycl_ext_oneapi_usm_raii.asciidoc

```diff
@@ -0,0 +1,135 @@
+SYCL specification refer to that revision.
+
+
+== Status
+
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+
+== Overview
+
+SYCL 2020 provides very C-like interfaces for USM allocation/deallocations
+routines. With in-order queues/USM and
+link:./sycl_ext_oneapi_enqueue_functions.asciidoc[sycl_ext_oneapi_enqueue_functions]
+as a future direction to write SYCL programs with as little overhead as possible
+we need usability improvements for the USM APIs that C++ programmers would
+expect.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_USM_RAII` to one of the values defined in
+the table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== USM Deleter
+
+This extension adds `usm_deleter` helper that can be used with `std::unique_ptr`
+or `std::shared_ptr` to manage deallocation of USM memory.
+
+[source, c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+class usm_deleter {
+public:
+  usm_deleter(const context &Ctx);
+  usm_deleter(const queue &q);
+
+  template <typename T> void operator()(T *Ptr) const;
+};
+} // sycl::ext::oneapi::experimental
+----
+
+=== Example usage
+
+A simple example of using this extension is below.
+
+```c++
+{
+  sycl::queue q;
+  using sycl::ext::oneapi::experimental::usm_deleter;
+  std::unique_ptr<int, usm_deleter> p{sycl::malloc_shared<int>(1, q), {q}};
+  q.single_task([=, p = p.get()]() { *p = 42; }).wait();
+}
+```
+
+== Issues/open questions/implementation details
+
+* link:https://en.cppreference.com/w/cpp/memory/shared_ptr/allocate_shared[`allocate_shared(const Alloc& alloc, Args&&... args )`]
+  seems to work with existing `usm_allocator`. However, it uses single
+  allocation for both the allocated object and auxilary data for
+  `std::shared_ptr`. Having that auxilary data residing in the same USM memory
+  might have unintended consequences depending on particular HW when copies are
+  created on host when the same memory page is being accessed on a device.
+* Providing `make_unique`/`make_shared`-like interface would lead to API bloat,
```

Comment:
I wouldn't call it API bloat. I do agree it is something which isn't required for the first revision.